web/src/main/java/org/springframework/security/web/server/authentication/AuthenticationWebFilter.java

```diff
@@ -51,18 +53,23 @@
  *     The {@link ReactiveAuthenticationManager} specified in
  *     {@link #AuthenticationWebFilter(ReactiveAuthenticationManager)} is used to perform authentication.
  * </li>
+ *<li>
+ *     The {@link ReactiveAuthenticationManagerResolver} specified in
+ *     {@link #AuthenticationWebFilter(ReactiveAuthenticationManagerResolver)} is used to resolve the appropriate
+ *     authentication manager from context to perform authentication.
+ * </li>
  * <li>
  *     If authentication is successful, {@link ServerAuthenticationSuccessHandler} is invoked and the authentication
  *     is set on {@link ReactiveSecurityContextHolder}, else {@link ServerAuthenticationFailureHandler} is invoked
  * </li>
  * </ul>
  *
  * @author Rob Winch
+ * @author Rafiullah Hamedy
  * @since 5.0
  */
 public class AuthenticationWebFilter implements WebFilter {
-
-	private final ReactiveAuthenticationManager authenticationManager;
+	private ReactiveAuthenticationManagerResolver<ServerHttpRequest> authenticationManagerResolver;
```

Comment:
Is there any reason this can't be `final`? Let's make it `final` if it can be.

Reply:
I don't see a reason why not - `final` it is 👍